activesupport/lib/active_support/core_ext/object/with_options.rb

```diff
@@ -75,8 +75,27 @@ class Object
   #     end
   #   end
   #
+  # When the block argument is omitted, the decorated Object instance is returned:
+  #
+  #   module MyStyledHelpers
+  #     def styled
+  #       with_options style: "color: red;"
+  #     end
+  #   end
+  #
+  #   # styled.link_to "I'm red", "/"
+  #   # #=> <a href="/" style="color: red;">I'm red</a>
+  #
+  #   # styled.button_tag "I'm red too!"
+  #   # #=> <button style="color: red;">I'm red too!</button>
+  #
   def with_options(options, &block)
     option_merger = ActiveSupport::OptionMerger.new(self, options)
-    block.arity.zero? ? option_merger.instance_eval(&block) : block.call(option_merger)
+
+    if block.nil?
+      option_merger
```

Comment:
@rafaelfranca does this change seem viable? Would it require promoting `ActiveSupport::OptionMerger` to public API?

Reply:
I think this changes make sense but I'd like to change `OptionMerger` to implement `respond_to_missing?` so people are able to know if the method exists in the context or not. What do you think?

Reply:
That's a good idea. Would delegating to `@context` be good enough?

```ruby
def respond_to_missing?(name, include_private = false)
  @context.respond_to_missing?(name, include_private)
end
```

Reply:
Yes. But I think it needs to delegate to `@context`'s `respond_to?`

Reply:
```diff
diff --git a/activesupport/lib/active_support/option_merger.rb b/activesupport/lib/active_support/option_merger.rb
index 696bbedd41..486b449696 100644
--- a/activesupport/lib/active_support/option_merger.rb
+++ b/activesupport/lib/active_support/option_merger.rb
@@ -30,5 +30,9 @@ def method_missing(method, *arguments, &block)
           @context.__send__(method, *arguments, &block)
         end
       end
+
+      def respond_to_missing?(method, include_private = false)
+        @context.__send__(:respond_to_missing?, method, include_private)
+      end
   end
 end
```

Or 

```diff
diff --git a/activesupport/lib/active_support/option_merger.rb b/activesupport/lib/active_support/option_merger.rb
index 696bbedd41..b96f790466 100644
--- a/activesupport/lib/active_support/option_merger.rb
+++ b/activesupport/lib/active_support/option_merger.rb
@@ -30,5 +30,9 @@ def method_missing(method, *arguments, &block)
           @context.__send__(method, *arguments, &block)
         end
       end
+
+      def respond_to_missing?(*arguments)
+        @context.__send__(:respond_to?, *arguments)
+      end
   end
 end

```

Reply:
Do we need to use `__send__`?. I'd go with the latter.

Reply:
Ah, you already changed. Perfect.